### DISCUSSION TITLE
Suggested adapters to migrate

### DISCUSSION THREAD

**Discussion by xefil** [Koenkk/zigbee2mqtt#8901] [General]
Hi all,
I'm using zigbee2mqtt since some years but from time to time I need to reset and re-pair the whole stuff.
Actually my setup is:
- zigbee2mqtt version 1.20.0-dev
- Coordinator on CC2531 (CC2531_SOURCE_ROUTING_20201128)
- Router on CC2530+CC2591 (CC2530_CC2591_router_2020_09_29)
- Router on CC2531
- more or less 25 devices

What drives me crazy is as soon I've problem with the Router on CC2530+CC2591 I cannot pair it, even with flash, reset, power on/off 3 times, and so one....
I'm planning to move all adapters to something new, possible more stable and easier to re-pair... a button? a led that blinks? CC2530 doesn't help to debug
Which device would yo uuse as coordinator? Which as router? Would it needed anyway?
And about the existing devices, could I leave them as router as well? I've seen those are on zstack1.2, new ones on zstack3 I've no idea if they are compatible.

Thanks, Simon

**Comment by francisp2** [comment]
Any on the list that is not a CC253x

https://www.zigbee2mqtt.io/information/supported_adapters.html

I have 3 CC2531 routers and 2 CC2530 routers, they work ok with a CC2652. Still need to flash my old coordinator (CC2530+CC2591+ESP01) to a router.

**Reply by xefil** [reply]
Hello @francisp2 , so you are using a coordinator device with zstack3 on your routers on zstack1.2, correct?
So your coordinator is a CC2652. Which CC2652 exactly do you have? Seems they are not all easy to find.
Even interested in the CC2530+CC2591+ESP01. I have some ESP8266 that I'm not using...
Sorry for all the questions :-)
Thanks, Simon

**Comment by francisp2** [comment]
I'm using a tubezb ethernet coordinator now, before the CC2530+CC2591+ESP01 as coordinator. 
see here : 
https://www.zigbee2mqtt.io/information/connecting_cc2530.html#via-an-esp8266-as-a-serial-to-wifi-bridge

**Comment by xefil** [comment]
Thanks @francisp2 for suggestions. Any idea someone how to debug `CC2530+CC2591` router? Is there a way to show via serial port (UART adapter COMx) to look what happens?
I've tried and get only:
`TexasInstruments SW5: Help       asInstruments SW5: Help ...`
@Koenkk any experience?
Or better throw away?

I would like to use a new coordinator (i.e. SMARTLIGHT CC2652P Zigbee USB dongle) but use my old sticks as router.
So, I ask another question. Can I use a CC2652P Coordinator with zstack3 and as router devices with zstack1.2 ?

Thanks all!

Simon

**Comment by francisp2** [comment]
> I would like to use a new coordinator (i.e. SMARTLIGHT CC2652P Zigbee USB dongle) but use my old sticks as router.
> So, I ask another question. Can I use a CC2652P Coordinator with zstack3 and as router devices with zstack1.2 ?

That is no problem. I have a CC2652P Coordinator with zstack3, 3 CC3531 routers with zstack 1.2 and 2 CC2530 routers with zstack 1.2

**Comment by xefil** [comment]
Hi @francisp2 , about CC2530 which firmware? And it's possible to use a serial usb adapter to show what's happening? It won't connect/join the coordinator again. I had a network outage (multiple times) and fear something went bad on the board, but the firmware gets uploaded corrctly.
Thanks for your help!
Simon

**Comment by francisp2** [comment]
Firmware can be found here : 

https://github.com/Koenkk/Z-Stack-firmware/tree/master/router/Z-Stack_Home_1.2/bin

**Comment by xefil** [comment]
The same I've installed @francisp2 . Do you know if there is a way to debug a cc2530 and see what it's doing? Hope in case @Koenkk has suggestions. Thanks!
Simon

**Comment by francisp2** [comment]
I have been trying to get my old coordinator (CC2530+CC2591) to join as a router, but no go :)
Flashed several times, with the koenkk  firmware, with the pvto firmware, won't pair.

**Reply by xefil** [reply]
Thanks for your update. I'm migrating to `CC2652P` as soon I'll get the hardware :-)
